### PR TITLE
Introduce an internal immutable frame.

### PR DESCRIPTION
**Pull request by ueshin**
This is part of #322.
This introduces `_InternalFrame` to store immutable structure of `sdf`, `data_column`, `index_map`, and `scol`.
We can still do further refactoring, but I'd leave it in the future PRs to keep each PR simpler.

### CODE REVIEW

File: databricks/koalas/internal.py [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+        """ Return the managed index names. """
+        return [index_name for _, index_name in self.index_map]
+
+    @property
+    def scol(self) -> Optional[spark.Column]:
+        """ Return the managed Spark Column. """
+        return self._scol
+
+    @lazy_property
+    def spark_dataframe(self) -> spark.DataFrame:
```
Comment (rxin) [inline databricks/koalas/internal.py]: spark_df?

**Comment by rxin** [comment]
Would it be useful to define a copy constructor, so we can do

```
new_idf = idf.copy(index_names=...)
```

File: databricks/koalas/internal.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/internal.py]: Renamed: `spark_dataframe` -> `spark_df`, `pandas_dataframe` -> `pandas_df`.

**Comment by ueshin** [comment]
We already have the copy constructor.

**Comment by ueshin** [comment]
Thanks! I will continue further refactoring.

**Pull request merged**